lib/infra/TranscodingEngine.js

```diff
@@ -0,0 +1,64 @@
+
+const Promise = require("bluebird");
+const exec = require('child_process').exec;
+const TranscodingEngineResponse = require('./TranscodingEngineResponse');
+const fs = require('fs');
+const path = require('path');
+
+/**
+ * class to handle transcoding
+ */
+class TranscodingEngine
+{
+
+    constructor(ffmpegPath = 'ffmpeg')

```

Comment:
I see that you use ffmpegPath without really depending on it to be ffmpeg - call it transcoderPath